### ISSUE TITLE
Age & History - Minor Corrections

### ISSUE THREAD

**Issue by mdsimpson42** [colouring-cities/colouring-britain#362]
## Core
Architectural Style/historical period
- [x] Data source options:
  - [x] Central government record/online database
  - [x] + Local government record/online database
  - [x] Open database (non-profit)
  - [x] + Open database (commercial)
  - [x] - Other independently managed database
  - [x] - Commercial Database
  - [x] + Website (Commercial)
  - [x] + Website (Non-profit)

~~Cladding, extensions and retrofits  
Date of significant extensions - add multiple?~~ (Created separate issue #367)

Building age/construction date
- [x] Construction start date
- [x] Estimated construction period - calculated

## Britain
- [x] The following should be removed from Core, as they are Britain specific.

... 

- [x] Remove one set of source links
- [x] Remove "specific" + "type" from historical source
- [x] Specific source type:
  - [x] Official conservation area appraisals (top)
  - [x] National heritage...
  - [x] Pevsner...
  - [x] Victoria County...
  - [x] Survey of London
  - [x] Local history publication (book/record)
  - [x] + Local history website
  - [x] Other website
  - [x] Other database or gazetter
  - [x] Historical Map
  - [x] Film/video
  - [x] Expert knowledge at bottom of list (above other)

**Comment by polly64** [comment]
can you also change text on the light dark button to say 

switch to light background
switch to dark background

**Comment by mdsimpson42** [comment]
Also, changing the data source options brings us back to the old problem of what do we do if there is existing data in the database? 

i.e. if someone has already put "Current government record/database" in as a value on some buildings and we rename it to "Central government record/online database", then the old data is effectively lost unless we update the records in the database. But we still haven't resolved how to do this while avoiding concerns about data provenance/history.

**Comment by mdsimpson42** [comment]
**_Building age/construction date_**

We may need some clarity on what the existing fields mean?
We already have:
- Year of construction of main building - should this be the year that construction was completed?
- Earliest possible start year - How is this different from "Construction start date" - this could be used to calculate the construction length, unless it means something else?
- Latest possible start year - ditto

**Comment by mdsimpson42** [comment]
**Historical source type**

The full list is now:

            "Official conservation area appraisals",
            "Pevsner Guides",
            "Victoria County History",
            "Survey of London",
            "Local history publication (book/record)",
            "Other publication",
            "Local history website",
            "Other website",
            "National Heritage List for England",
            "Other database or gazetteer",
            "Historical map",
            "Film/Video",
            "Other archive document",
            "Expert knowledge of building",
            "Expert estimate from image",
            "Other"

I just wanted to confirm that that is correct before committing it.

**Comment by polly64** [comment]
great yes all ok

**Comment by polly64** [comment]
@mdsimpson42 can we correct this to minimum 1
<img width="434" alt="Screenshot 2024-10-15 at 16 25 51" src="https://github.com/user-attachments/assets/b6764a77-1b8d-437f-a7b2-44e42fd6f77e">

**Comment by mdsimpson42** [comment]
I've set it so that the minimum value of the duration is 1. That does mean you get the same value if the completion date is 2007 or 2008 in the above case, but that should be OK?

I've also added some logic so that the start year can't be larger than the end year (and the end year can't be smaller than the start year), at least when using the arrows.

---

I will check these changes into Core. They will appear in a future deployment.
